python/cuml/test/test_pickle.py

```diff
@@ -71,17 +65,12 @@
     "UMAP": lambda: cuml.UMAP()
 }
 
-rf_models = {
-    "rfc": lambda: cuml.RandomForestClassifier(),
-    "rfr": lambda: cuml.RandomForestRegressor()
-}
+rf_models = ModuleConfig(module=cuml.ensemble)
+rf_models = rf_models.get_models()
 
-k_neighbors_models = {
-    "KNN-Classifer": lambda n_neighbors=10: cuml.neighbors.
-    KNeighborsClassifier(n_neighbors=n_neighbors),
-    "KNN-Regressor": lambda n_neighbors=10: cuml.neighbors.
-    KNeighborsRegressor(n_neighbors=n_neighbors)
-}
+k_neighbors_config = ModuleConfig(module=cuml.neighbors, exclude_classes=[
+    cuml.neighbors.NearestNeighbors])
+k_neighbors_models = k_neighbors_config.get_models()
```

Comment:
It seems like svm models are missing, no?

Reply:
SVM models are tested in:
https://github.com/rapidsai/cuml/blob/6c102c0b15a3fc29eb8654fb1ac04a57f7d036a5/python/cuml/test/test_pickle.py#L478-L519
They do not use the ModuleConfig because at the moment they have specific tests and there is no need to group them with ModuleConfig. Although maybe they should have some common tests are therefore ModuleConfig would be suitable for them as well.

Reply:
I think it will be helpful to ensure we have at least a couple of tests that run on absolutely every model, even if they have more specific tests too. If we need to use pytest.xfail for some weird one, we can do that where necessary. Otherwise, we still run the risk of missing key functionality for some class.

Reply:
I fully agree, every model should pass the normal test, with more complex models having additional tests if needed

Reply:
I definitely agree with both of you. So I tried running the 2 "normal" tests:
https://github.com/Garfounkel/cuml/blob/7e88db51f9f8f421374be96ea882b69bfcdc8d8e/python/cuml/test/test_pickle.py#L302-L323
It's definitely a good thing since it adds 16 tests right off the bat. Problem is 9 of them are failing. I started to look at `test_unfit_pickle[ARIMAModel]` but I could not understand why it fails yet. I changed back the PR into [WIP] and I'll continue to investigate those 9 tests tomorrow to double check if they should xfail or not.

Reply:
So I was able to fix one of the 9 failing tests, here are the rest:
```
ARIMAModel: xfail
    unfit_pickle -> ValueError: ctypes objects containing pointers cannot be pickled
    unfit_clone  -> TypeError: no default __reduce__ due to non-trivial __cinit__

ExponetiolSmoothing: xfail
    unfit_clone -> TypeError: no default __reduce__ due to non-trivial __cinit__
    
MBSGDClassifier: xfail
    unfit_clone -> TypeError: no default __reduce__ due to non-trivial __cinit__

MBSGDRegressor: xfail
    unfit_clone -> TypeError: no default __reduce__ due to non-trivial __cinit__
    
KalmanFilter: xfail
    unfit_pickle -> ValueError: ctypes objects containing pointers cannot be pickled
    unfit_clone  -> ValueError: ctypes objects containing pointers cannot be pickled
    
ForestInference: xfail
    unfit_pickle -> TypeError: no default __reduce__ due to non-trivial __cinit__
```
I took the liberty to make them xfail because they don't seem like trivial issues, especially `ValueError: ctypes objects containing pointers cannot be pickled`. 

As for `TypeError: no default __reduce__ due to non-trivial __cinit__`, I'm not sure how to fix it but it looks like that would be feasible? What do you think @JohnZed @dantegd? Should I file an issue for those?

Reply:
@Garfounkel yes please fill an issue to track them. @JohnZed most of these, at least the ctypes pointers, will be solved by adopting #1635

Reply:
Filed issue at #1651.